types/url-format-lax/index.d.ts

```diff
@@ -0,0 +1,10 @@
+// Type definitions for url-format-lax 2.0
+// Project: https://github.com/sindresorhus/url-format-lax
+// Definitions by: Junxiao Shi <https://github.com/yoursunny>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node" />
+
+import { UrlObject } from "url";
+
+export default function urlFormat(url: UrlObject): string;
```

Comment:
can you consider to retype used props from NodeJS shape to avoid direct dependency here to NodeJS types (it's only for DT, to save time when CI tests multiple DT types, if you stick with NodeJS, your package would become indirect dependency for DT types for NodeJS). It won't impact runtime TS, as TS just describe shapes:
```suggestion

// used shape of UrlObject from NodeJS
// to remove introducing dependency for NodeJS DT CI
interface Url {
    host?: string | null | undefined;
    hostname?: string | null | undefined;
    port?: string | number | undefined
}

export default function urlFormat(url: Url): string;
```

Reply:
Yes I thought about that but noticed the following problem.

One of the examples given on https://www.npmjs.com/package/url-format-lax is:
```js
urlFormat({protocol: 'https', hostname: 'google.com', port: '123'});
```

If I declare `interface UrlObject` like suggested, this example would fail to compile due to having unrecognized property.

Reply:
OK, can you add a passhtrough?
```ts
interface Url {
    host?: string | null | undefined;
    hostname?: string | null | undefined;
    port?: string | number | undefined;
    [other: string]: any;
}
```
(or that's ok to copy whatever can be passed there, I"ve picked those after seeing method body)